docs/vagrant.md

```diff
@@ -7,25 +7,155 @@ You can use following guide to setup a cluster in a virtual machine.
 * Vagrant
 
 ### Preparation
-* bin/build-release.sh must rename _release/apache-storm-${RELEASE} to _release/storm-mesos-${RELEASE}
-* sudo echo "192.168.50.101 master" >> /etc/hosts
-* sudo echo "192.168.50.102 slave"  >> /etc/hosts
+1. Add name lookup entries for the 2 VMs:
+
+  ```shell
+  sudo sh -c 'echo "192.168.50.101 master" >> /etc/hosts'
+  sudo sh -c 'echo "192.168.50.102 slave"  >> /etc/hosts'
+  ```
+
+2. Update `storm.yaml`, specifically ensuring you set these fields:
+
+  ```shell
+  mesos.master.url: "zk://master:2181/mesos"
+  storm.zookeeper.servers:
+  - "master"
+  # Modify package version as appropriate
+  mesos.executor.uri: "file:///usr/local/storm/storm-mesos-0.9.6.tgz"
+  ```
 
 ### Run
 
 ```shell
 bin/build-release.sh
-cp vagrant/Vagrantfile .
 ```
 
 To start the cluster run following:
 ```shell
 vagrant up
 ```
-At this point the VM will have two node mesos cluster running, node 1 (master) will have a mesos master and slave running and node 2 (slave) will have only a mesos slave running. Docker has been installed as well.
+This will bring up a 2-node mesos cluster:
+* node 1 (master) will have a mesos master and slave running.
+* node 2 (slave) will have only a mesos slave running.
+Docker has been installed as well.
 
-To ssh in the cluster run following:
+To ssh into the cluster hosts, run the following commands:
 ```shell
 vagrant ssh master
 vagrant ssh slave
-``` 
+```
+
+Then you can submit a storm topology to the nimbus from outside the guest VMs like so:
+```shell
+cp my-topology.jar vagrant/.
+vagrant ssh master -c "/vagrant/vagrant/submit-topology.sh my-topology.jar foo.bar.TopologyMain optionalArgument1 optionalArgument2 ..."
+```
+
+### Cluster Web Interfaces
+
+* [Storm Nimbus UI](http://master:8080)
+* [Mesos UI](http://master:5050)
+
+### Debugging:
+WARNING: If you do not correctly configure `storm.yaml` by setting all of the mandatory options specified in this project's README.md, then various things will break, and it may be hard to figure out exactly what is at fault.
+
+Various logs can be access from within the VM:
+```shell
+# Storm Nimbus:
+/vagrant/_release/storm-mesos-*/logs/nimbus.log
+
+# Storm UI:
+/vagrant/_release/storm-mesos-*/logs/ui.log
+
+# mesos-master
+/var/log/mesos/mesos-master.INFO
+
+# mesos-slave
+/var/log/mesos/mesos-slave.INFO
+
+# For below mesos-executor related logs, replace the following variables:
+#  SLAVE_ID - the ID of the mesos-slave instance
+#  FRAMEWORK_ID - the ID of the framework you are running.  Just use the newest dir in that location.
+#  TOPOLOGY_ID - the ID of the storm topology you are running
```

Comment:
It might be worth mentioning that this is the ID that Storm reports in its UI, not simply the name of the topology that you submitted.